src/main/java/appeng/client/render/tesr/CableBusFastTESR.java

```diff
@@ -0,0 +1,306 @@
+		if( renderState == null )
+		{
+			return;
+		}
+
+		boolean thinFacades = FacadeBuilder.isUseThinFacades( renderState.getBoundingBoxes() );
+		long rand = MathHelper.getPositionRandom( pos );
+
+		renderState.getFacades().forEach( ( side, state ) ->
+		{
+			AxisAlignedBB facadeBox = FacadeBuilder.getFacadeBox( side, thinFacades );
+			AEAxisAlignedBB cutOutBox = FacadeBuilder.getCutOutBox( facadeBox, renderState.getBoundingBoxes() );
+			List<BakedQuad> quads = this.addFacade(
+					MinecraftForgeClient.getRenderLayer(),
+					renderState.getFacades(),
+					side,
+					cutOutBox,
+					thinFacades,
+					rand );
+
+			buffer.setTranslation( x, y, z );
+			VertexLighterFlat l = new VertexLighterSmoothAo( Minecraft.getMinecraft().getBlockColors() );
+			l.setParent( new VertexBufferConsumer( buffer ) );
+			l.setWorld( world );
+			l.setBlockPos( pos );
+			l.setState( state.getSourceBlock() );
+			l.updateBlockInfo();
+
+			for( BakedQuad q : quads )
+			{
+				q.pipe( l );
+			}
+		} );
+	}
+
+	private static CableBusRenderState getRenderingState( final IBlockAccess world, BlockPos pos )
+	{
+		IBlockState blockState = world.getBlockState( pos );
+		try
+		{
+			blockState = blockState.getActualState( world, pos );
+		}
+		catch( Exception e )
+		{
+			// ignore
+		}
+
+		try
+		{
+			blockState = blockState.getBlock().getExtendedState( blockState, world, pos );
+		}
+		catch( Exception e )
+		{
+			// ignore
+		}
+
+		if( blockState == null || !( blockState instanceof IExtendedBlockState ) )
+		{
+			return null;
+		}
+
+		IExtendedBlockState extendedBlockState = (IExtendedBlockState) blockState;
+		return extendedBlockState.getValue( BlockCableBus.RENDER_STATE_PROPERTY );
+	}
+
+	private List<BakedQuad> addFacade( BlockRenderLayer layer, Map<EnumFacing, FacadeRenderState> facades, EnumFacing side, AEAxisAlignedBB busBounds, boolean thinFacades, long rand )
+	{
+		ArrayList<BakedQuad> ret = new ArrayList<>();
+
+		CubeBuilder builder = new CubeBuilder( DefaultVertexFormats.ITEM, ret );
+		FacadeRenderState facadeState = facades.get( side );
+		IBlockState blockState = facadeState.getSourceBlock();
+
+		builder.setDrawFaces( EnumSet.allOf( EnumFacing.class ) );
+
+		// Reset to no color multiplicator
+		builder.setColorRGB( 0xFFFFFF );
+		builder.useStandardUV();
+
+		// Do not add the translucent facade in any other layer than translucent
+		boolean translucent = AEApi.instance().partHelper().getCableRenderMode().transparentFacades;
+		if( translucent && layer != BlockRenderLayer.TRANSLUCENT )
+		{
+			return ret;
+		}
+
+		final float thickness = thinFacades ? 1 : 2;
+
+		BlockRendererDispatcher blockRendererDispatcher = Minecraft.getMinecraft().getBlockRendererDispatcher();
+		IBakedModel blockModel = blockRendererDispatcher.getModelForState( blockState );
+
+		final int color;
+		if( translucent )
+		{
+			color = 0x4CFFFFFF;
+		}
+		else
+		{
+			color = 0xFFFFFFFF;
+		}
+
+		// TODO: Cache this
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)